express-delete/server.js

```diff
@@ -0,0 +1,45 @@
+import express from 'express';
+const app = express();
+
+const grades = {
+  12: {
+    id: 12,
+    name: 'Tim Berners-Lee',
+    course: 'HTML',
+    score: 95
+  },
+  47: {
+    id: 47,
+    name: 'Brendan Eich',
+    course: 'JavaScript',
+    score: 100
+  },
+  273: {
+    id: 273,
+    name: 'Forbes Lindsay',
+    course: 'JavaScript',
+    score: 92
+  }
+};
+
+app.get('/api/grades', (req, res) => {
+  const gradesArray = [];
+  for (const id in grades) {
+    gradesArray.push(grades[id]);
+  }
+  res.json(gradesArray);
+});
+
+app.delete('/api/grades/:id', (req, res) => {
+  const id = req.params.id;
+  if (grades[id]) {
+    delete grades[id];
+    res.sendStatus(204);
+  } else {
+    res.status(404);
+  }
```

Comment:
Please make sure a response is sent to each request, even if that response is an error. The `status` method only sets the status code / message, it doesn't send a response.

Reply:
Fixed!